app/controllers/line_items_visits_controller.rb

```diff
@@ -0,0 +1,118 @@
+    arm               = line_items_visit.arm
+    line_items_visits = arm.line_items_visits.eager_load(line_item: [:admin_rates, service: [:pricing_maps, organization: [:pricing_setups, parent: [:pricing_setups, parent: [:pricing_setups, parent: :pricing_setups]]]], service_request: :protocol])
+    visit_groups      = arm.visit_groups.paginate(page: page.to_i, per_page: VisitGroup.per_page).eager_load(visits: { line_items_visit: { line_item: [:admin_rates, service: [:pricing_maps, organization: [:pricing_setups, parent: [:pricing_setups, parent: [:pricing_setups, parent: :pricing_setups]]]], service_request: :protocol] } })
+
+    if line_items_visit.update_attributes(line_items_visit_params)
+      unless params[:portal] == 'true'
+        line_items_visit.sub_service_request.update_attribute(:status, 'draft')
+        @service_request.update_attribute(:status, 'draft')
+      end
+      render json: { 
+               total_per_patient: render_to_string(partial: 'service_calendars/master_calendar/pppv/total_per_patient', locals: { liv: line_items_visit }),
+               total_per_study: render_to_string(partial: 'service_calendars/master_calendar/pppv/total_per_study', locals: { liv: line_items_visit }),
+               max_total_direct: render_to_string(partial: 'service_calendars/master_calendar/pppv/totals/max_total_direct_per_patient', locals: { arm: arm, visit_groups: visit_groups, line_items_visits: line_items_visits, tab: tab, page: page }),
+               max_total_per_patient: render_to_string(partial: 'service_calendars/master_calendar/pppv/totals/max_total_per_patient', locals: { arm: arm, visit_groups: visit_groups, line_items_visits: line_items_visits, tab: tab, page: page }),
+               total_costs: render_to_string(partial: 'service_calendars/master_calendar/pppv/totals/total_cost_per_study', locals: { arm: arm, line_items_visits: line_items_visits, tab: tab }),
+               success: true
+             }
+    else
+      render json: line_items_visit.errors, status: :unprocessable_entity
+    end
+
+  end
+
+  def destroy
+    @line_items_visit = LineItemsVisit.find(params[:id])
+    @sub_service_request = @line_items_visit.line_item.sub_service_request
+    @service_request = @sub_service_request.service_request
+    @selected_arm = @service_request.arms.first
+    line_item = @line_items_visit.line_item
+    @line_items = @sub_service_request.line_items
+
+    ActiveRecord::Base.transaction do
+      if @line_items_visit.destroy
+        line_item.destroy unless line_item.line_items_visits.count > 0
+        # Have to reload the service request to get the correct direct cost total for the subsidy
+        @service_request = @sub_service_request.service_request
+        render 'dashboard/sub_service_requests/add_line_item'
+      end
+    end
+  end
+
+  private
+
+  def line_items_visit_params
+    params.require(:line_items_visit).permit(:subject_count)
+  end
+
+  private
+
+  def in_dashboard?
```

Comment:
You already have these methods in ApplicationController. They should be available since you are inheriting from that controller.